addons/binding/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/handler/FoobotHandler.java

```diff
@@ -0,0 +1,306 @@
+                response = request.send();
+                logger.debug("foobotResponse = {}", response);
+
+                if (response.getStatus() != 200) {
+                    errorMsg = response.getContentAsString();
+                    updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.COMMUNICATION_ERROR, errorMsg);
+                    return;
+                }
+                // Map the JSON response to list of objects
+                Type listType = new TypeToken<ArrayList<FoobotDevice>>() {
+                }.getType();
+                String userDevices = response.getContentAsString();
+                devices = GSON.fromJson(userDevices, listType);
+            } catch (ExecutionException | TimeoutException | InterruptedException ex) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, ex.getMessage());
+                return;
+            }
+        } else {
+            devices = this.account.getAssociatedDevices();
+        }
+
+        if (devices == null) {
+            errorMsg = "No devices found on this account.";
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            return;
+        }
+
+        for (FoobotDevice ob : devices) {
+            // Compare the mac address to each record in order to fetch the UUID of the current device.
+            if (config.mac.equals(ob.getMac())) {
+                this.device = ob;
+                break;
+            }
+        }
+
+        if (this.device == null) {
+            errorMsg = "No device found for this MAC address.";
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+        startAutomaticRefresh();
+    }
+
+    /**
+     * Start the job refreshing the Sensor data from Foobot device
+     */
+    private void startAutomaticRefresh() {
+        int delay = (config.refreshIntervalInMinutes != null) ? config.refreshIntervalInMinutes.intValue()
+                : FoobotBindingConstants.DEFAULT_REFRESH_PERIOD_MINUTES;
+        refreshDataJob = scheduler.scheduleWithFixedDelay(() -> {
+            // Request new sensor data from foobot.io service
+            foobotData = updateFoobotData();
+
+            // Update all channels from the updated Sensor data
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId(), foobotData);
+            }
+        }, 0, delay, TimeUnit.MINUTES);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Foobot handler.");
+        if (refreshDataJob != null && !refreshDataJob.isCancelled()) {
+            refreshDataJob.cancel(true);
+            refreshDataJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(@NonNullByDefault ChannelUID channelUID, @NonNullByDefault Command command) {
```

Comment:
Please annotate the class `@NonNullByDefault` and apply `@Nullable` where needed, see also [the coding guidelines](https://www.openhab.org/docs/developer/development/guidelines.html#a-code-style)